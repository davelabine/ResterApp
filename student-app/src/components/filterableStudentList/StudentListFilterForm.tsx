import * as React from 'react';
import './FilterableStudentList.css';
import { Form, FormGroup, ControlLabel, FormControl, FormControlProps, Button } from 'react-bootstrap';
import { EditStudentModal } from '../editStudentForm/editStudentModal';
import { StudentData } from '../../types';

export interface StudentListFilterFormProps {
    filter: string;
    onFilterStudents?: (filter: String) => void;
    onAddStudent?: (student: StudentData) => void;
}

export interface StudentListFilterState {
    show: boolean;
}

export class StudentListFilterForm extends React.Component<StudentListFilterFormProps, StudentListFilterState> {
    constructor(props: StudentListFilterFormProps) {
        super(props);

        this.state = { show: false };

        this.handleFormFilterChange = this.handleFormFilterChange.bind(this);
        this.onShowModal = this.onShowModal.bind(this);
        this.onHideModal = this.onHideModal.bind(this);
    }

    public handleFormFilterChange(e: React.FormEvent<FormControlProps>): void {
        let f = e.currentTarget.value as string;
        if (this.props.onFilterStudents) {
            this.props.onFilterStudents(f);
        }
    }

    public onShowModal() {
        this.setState({ show: true});
    }

    public onHideModal() {
        this.setState({ show: false});
    }

    public render() {
        const student = new StudentData();
        return (
            <Form inline={true}>
                <FormGroup controlId="formFilterStudents">
                    <ControlLabel>Search Last Name:</ControlLabel>
                    <FormControl
                        type="text"
                        onChange={this.handleFormFilterChange}
                        value={this.props.filter}
                    />
                    <Button 
                        className="pull-right"
                        type="button"
                        bsStyle="default"
                        onClick={this.onShowModal}
                    >Add Student
                    </Button>
                </FormGroup>
                <EditStudentModal 
                    title="Add Student"
                    submitButtonText="Add Student"
                    initialStudent={student}
                    show={this.state.show}
                    onHide={this.onHideModal}
                    onSubmit={this.props.onAddStudent}
                />
            </Form>
        );
    }
}

export default StudentListFilterForm;
