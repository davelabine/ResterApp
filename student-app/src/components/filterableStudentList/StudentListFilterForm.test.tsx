import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { StudentListFilterForm } from './StudentListFilterForm';

describe('FilterableStudentList', () => {
    it('renders without crashing', () => {
      mount(<StudentListFilterForm filter="" onFilterChange={jest.fn()}/>);
    });
    
    it('triggers callback when input is changed', () => {
      let mockOnFilterChange = jest.fn();
      let form = shallow(<StudentListFilterForm filter="" onFilterChange={mockOnFilterChange}/>);
      let input = form.find('input');
      expect(input.length).toEqual(1);
      input.simulate('change', {target: {value: 'abc'}});
      expect(mockOnFilterChange).toHaveBeenCalled();
    });
});