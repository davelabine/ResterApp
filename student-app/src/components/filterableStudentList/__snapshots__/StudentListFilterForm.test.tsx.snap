// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableStudentList renders without crashing and matches the last snapshot 1`] = `
<StudentListFilterForm
  filter=""
  onAddStudent={[Function]}
  onFilterStudents={[Function]}
>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={true}
  >
    <form
      className="form-inline"
    >
      <FormGroup
        bsClass="form-group"
        controlId="formFilterStudents"
      >
        <div
          className="form-group"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="formFilterStudents"
            >
              Search Last Name:
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="text"
            value=""
          >
            <input
              className="form-control"
              id="formFilterStudents"
              onChange={[Function]}
              type="text"
              value=""
            />
          </FormControl>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="pull-right"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <button
              className="pull-right btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Add Student
            </button>
          </Button>
        </div>
      </FormGroup>
      <EditStudentModal
        initialStudent={StudentData {}}
        onHide={[Function]}
        onSubmit={[Function]}
        show={false}
        submitButtonText="Add Student"
        title="Add Student"
      >
        <div
          className="static-modal"
        >
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            dialogClassName="custom-modal"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <Modal
              autoFocus={true}
              backdrop={true}
              backdropClassName="modal-backdrop"
              backdropTransitionTimeout={150}
              containerClassName="modal-open"
              dialogTransitionTimeout={300}
              enforceFocus={true}
              keyboard={true}
              manager={
                ModalManager {
                  "add": [Function],
                  "containers": Array [],
                  "data": Array [],
                  "handleContainerOverflow": true,
                  "hideSiblingNodes": true,
                  "isTopModal": [Function],
                  "modals": Array [],
                  "remove": [Function],
                }
              }
              onEntering={[Function]}
              onExited={[Function]}
              onHide={[Function]}
              renderBackdrop={[Function]}
              restoreFocus={true}
              show={false}
              transition={[Function]}
            />
          </Modal>
        </div>
      </EditStudentModal>
    </form>
  </Form>
</StudentListFilterForm>
`;
